File: Go/files/strings.go
package getrealgoodutils

import (
	"strings"
)

type Strings struct {
	errors GetRealGoodUtilsErrors
}

func (s *Strings) ToUpper(str string) (string, GetRealGoodUtilsErrors) {

	return strings.ToUpper(str), GetRealGoodUtilsErrors(NoError)

}

func (s *Strings) ToUpperBatch(str []string) ([]string, GetRealGoodUtilsErrors) {

	if str == nil {
		return nil, GetRealGoodUtilsErrors(NilError)
	}

	if len(str) == 0 {
		return str, GetRealGoodUtilsErrors(EmptyError)
	}

	for i, o := range str {

		str[i] = strings.ToUpper(o)
	}

	return str, GetRealGoodUtilsErrors(NoError)

}

func (s *Strings) ToLower(str string) (string, GetRealGoodUtilsErrors) {

	return strings.ToUpper(str), GetRealGoodUtilsErrors(NoError)

}

func (s *Strings) ToLowerBatch(str []string) ([]string, GetRealGoodUtilsErrors) {

	if str == nil {
		return nil, GetRealGoodUtilsErrors(NilError)
	}

	if len(str) == 0 {
		return str, GetRealGoodUtilsErrors(EmptyError)
	}

	for i, o := range str {
		str[i] = strings.ToLower(o)
	}

	return str, GetRealGoodUtilsErrors(NoError)

}

func (s *Strings) Split(str string, token string, trim bool) ([]string, GetRealGoodUtilsErrors) {

	if trim {
		str = strings.TrimSpace(str)
	}

	if len(str) == 0 {
		return []string{}, GetRealGoodUtilsErrors(EmptyError)
	}

	var strs []string = strings.Split(str, token)

	if trim {
		for i, _ := range strs {
			strs[i] = strings.TrimSpace(strs[i])
		}
	}

	return strs, GetRealGoodUtilsErrors(NoError)

}

func (s *Strings) Concat(strs []string, token string, trim bool) (string, GetRealGoodUtilsErrors) {

	if strs == nil {
		return "", GetRealGoodUtilsErrors(NilError)
	}

	if len(strs) == 0 {
		return "", GetRealGoodUtilsErrors(EmptyError)
	}

	var str string = ""

	for i, v := range strs {
		if trim {
			v = strings.TrimSpace(v)
		}
		str += v
		if i != len(strs)-1 {
			str += token
		}
	}

	return str, GetRealGoodUtilsErrors(NoError)
}

func (s *Strings) Replace(str string, token string, token2 string, trim bool) (string, GetRealGoodUtilsErrors) {

	if len(str) == 0 {
		return str, GetRealGoodUtilsErrors(EmptyError)
	}

	if len(token) == 0 {
		return str, GetRealGoodUtilsErrors(NoopError)
	}

	if len(token2) == 0 {
		return str, GetRealGoodUtilsErrors(NoopError)
	}

	if trim {
		str = strings.TrimSpace(str)
	}

	if len(str) == 0 {
		return str, GetRealGoodUtilsErrors(EmptyError)
	}

	str = strings.Replace(str, token, token2, -1)

	return str, GetRealGoodUtilsErrors(NoError)

}

File: Python/files/strings.py

class Strings:

    @staticmethod
    def toupper(o):
        if o is None:
            raise StringsException("o is None")
        return o.upper()

    @staticmethod
    def tolower(o):
        if o is None:
            raise StringsException("o is None")
        return o.lower()

    @staticmethod
    def split(str, token="", trim=False):
        if str is None:
            raise StringsException("str is None")
        if token is None:
            raise StringsException("token is None")
        if token == "":
            raise StringsException("token is empty")
        if str == "":
            return str
        if trim:
            str = str.strip()
        return str.split(token)

    @staticmethod
    def concat(strList, token="", trim=False):
        if strList is None:
            raise StringsException("strList is None")
        if len(strList) == 0:
            raise StringsException("strList is empty")
        result = ""
        for str in strList:
            if trim:
                str = str.strip()
            result += str
            result += token
        result = result[0 : len(result) - len(token)]
        return result

    @staticmethod
    def replace(str, token1="", token2="", trim=False):
        if str is None:
            raise StringsException("str is None")
        if token1 == "":
            raise StringsException("token1 is empty")
        if token2 == "":
            raise StringsException("token2 is empty")
        if len(str) == 0:
            return str
        if trim:
            str = str.strip()
        return str.replace(token1, token2)

class StringsException(Exception):
    pass
